src/wmtk/simplex/SimplexCollection.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <vector>
+#include <wmtk/Mesh.hpp>
+#include "Simplex.hpp"
+#include "internal/SimplexEqualFunctor.hpp"
+#include "internal/SimplexLessFunctor.hpp"
+
+namespace wmtk::simplex {
+class SimplexCollection
+{
+public:
+    SimplexCollection(const Mesh& mesh)
+        : m_mesh{mesh}
+        , m_simplex_is_less(mesh)
+        , m_simplex_is_equal(mesh)
+    {}
+
+    /**
+     * @brief Return const reference to the simplex vector.
+     */
+    const std::vector<Simplex>& simplex_vector() const { return m_simplices; }
+    /**
+     * @brief Return vector of all simplices of the requested type.
+     */
+    std::vector<Simplex> simplex_vector(const PrimitiveType& ptype) const;
+
+    /**
+     * @brief Add simplex to the collection.
+     *
+     * There is no sorting or any check if the vertex already exists
+     */
+    void add(const Simplex& simplex);
+
+    void add(const SimplexCollection& simplex_collection);
+
+    /**
+     * @brief Sort simplex vector and remove duplicates.
+     */
+    void sort_and_clean();
```

Comment:
i think this should be private - each of the operations should by default default make sure it's cleaned/sorted?

Reply:
I'd like this to be public for the same reason as before. Sometimes it's better to sort things manually. An example is if you need a simplex with its boundary:
```
Simplex s;
SimplexCollection bd = simplex_boundary(mesh, s, false);
bd.add(s);
bd.sort_and_clean();
```